screen off function -S does not work Android Q beta 4 devices

```
arne@styx:~% scrcpy -S       
/usr/local/Cellar/scrcpy/1.9/share/scrcpy/scrcpy-server.jar: 1 file pushed. 3.4 MB/s (22587 bytes in 0.006s)
2019-06-12 16:01:31.151 scrcpy[41297:3693520] INFO: Initial texture: 1440x2880
[server] ERROR: Exception on thread Thread[Thread-1,5,main]
java.lang.AssertionError: java.lang.NoSuchMethodException: android.view.SurfaceControl.getBuiltInDisplay [int]
	at com.genymobile.scrcpy.wrappers.SurfaceControl.getBuiltInDisplay(SurfaceControl.java:82)
	at com.genymobile.scrcpy.Device.setScreenPowerMode(Device.java:164)
	at com.genymobile.scrcpy.Controller.handleEvent(Controller.java:113)
	at com.genymobile.scrcpy.Controller.control(Controller.java:73)
	at com.genymobile.scrcpy.Server$1.run(Server.java:45)
	at java.lang.Thread.run(Thread.java:919)
Caused by: java.lang.NoSuchMethodException: android.view.SurfaceControl.getBuiltInDisplay [int]
	at java.lang.Class.getMethod(Class.java:2072)
	at java.lang.Class.getMethod(Class.java:1693)
	at com.genymobile.scrcpy.wrappers.SurfaceControl.getBuiltInDisplay(SurfaceControl.java:80)
	... 5 more
```

Seems the API you are using has disappeared/changed in Android Q :(

If it is not possible to have the screen off feature in Android Q, maybe instead print a more understandable warning like "API to turn off screen not found in system image"?

--- a/server/src/main/java/com/genymobile/scrcpy/wrappers/SurfaceControl.java
+++ b/server/src/main/java/com/genymobile/scrcpy/wrappers/SurfaceControl.java
@@ -2,6 +2,7 @@ package com.genymobile.scrcpy.wrappers;
 
 import android.annotation.SuppressLint;
 import android.graphics.Rect;
+import android.os.Build;
 import android.os.IBinder;
 import android.view.Surface;
 
@@ -77,7 +78,13 @@ public final class SurfaceControl {
 
     public static IBinder getBuiltInDisplay(int builtInDisplayId) {
         try {
-            return (IBinder) CLASS.getMethod("getBuiltInDisplay", int.class).invoke(null, builtInDisplayId);
+            // Android Q does not have this method anymore but has a
+            // replacement.
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
+                return (IBinder) CLASS.getMethod("getBuiltInDisplay", int.class).invoke(null, builtInDisplayId);
+            } else {
+                return (IBinder) CLASS.getMethod("getPhysicalDisplayToken", long.class).invoke(null, builtInDisplayId);
+            }
         } catch (Exception e) {
             throw new AssertionError(e);
         }

